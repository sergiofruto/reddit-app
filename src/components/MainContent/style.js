import styled from 'styled-components';

export const Post = styled.article`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 32px;
`;

export const PostAuthor = styled.h1`
  margin: 0 0 32px;
  text-align: center;
`;

export const PostImage = styled.img`
  max-width: 100%;
  margin: 0 auto 32px;
`;

export const PostTitle = styled.h2`
  margin: 0 0 8px;
  text-align: center;
`;

export const Placeholder = styled.span`
  display: block;
  padding-top: 2rem;
  text-align: center;
`;

