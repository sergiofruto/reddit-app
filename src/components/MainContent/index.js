import React from 'react';
import { useSelector } from "react-redux";
import { Post, PostAuthor, PostImage, PostTitle, Placeholder } from './style';

const MainContent = () => {
  const selectedPost = useSelector(state => state.reddit.selectedPost);

  return (
    <section>
      {(selectedPost === undefined || selectedPost.length === 0) ? <Placeholder>Select a post to read</Placeholder> : (
        <Post>
          <PostAuthor>{selectedPost.author}</PostAuthor>
          {selectedPost.thumbnail && <PostImage src={selectedPost.thumbnail} alt={selectedPost.title}/>}
          {/* {selectedPost.preview.images && <PostImage src={selectedPost.preview.images[0].source.url} alt={selectedPost.title} />} */}
          <PostTitle>{selectedPost.title}</PostTitle>
        </Post>
      )}
    </section>
  );
};

export default MainContent;