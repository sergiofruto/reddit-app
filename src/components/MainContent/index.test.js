import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import MainContent from '.';

let mockData = require('./../../__tests__/mock.json');
const mockStore = configureStore([]);

describe('MainContent React-Redux Component', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      reddit: mockData.reddit,
    });

    store.dispatch = jest.fn();

    component = renderer.create(
      <Provider store={store}>
        <MainContent post={mockData.reddit.posts[0].data} />
      </Provider>
    );
  });

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});