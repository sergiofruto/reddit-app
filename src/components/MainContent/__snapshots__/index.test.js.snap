// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContent React-Redux Component should render with given state from Redux store 1`] = `
<section>
  <span
    className="sc-eCssSg jxVGDZ"
  >
    Select a post to read
  </span>
</section>
`;
